cub/device/device_adjacent_difference.cuh

```diff
@@ -109,53 +109,35 @@ private:
                      std::size_t &temp_storage_bytes,
                      InputIteratorT d_input,
                      OutputIteratorT d_output,
-                     std::size_t num_items,
+                     NumItemsT num_items,
                      DifferenceOpT difference_op,
                      cudaStream_t stream,
                      bool debug_synchronous)
   {
-    const auto uint64_threshold = static_cast<std::size_t>(
-      THRUST_NS_QUALIFIER::detail::integer_traits<
-        THRUST_NS_QUALIFIER::detail::int32_t>::const_max);
+    static_assert(
```

Comment:
Instead of reimplementing this logic for each Device algorithm, let's take the `ChooseOffsetT` helper that Andy added to `device_radix_sort.cuh`, move it to `cub/detail/choose_offset.cuh`, and reuse it here.

Reply:
Good idea, thanks! I've made one more change in the extracted `ChooseOffset`, though. I've replaced `uint32_t` with `std::uint32_t`.